[2022-10-31T20:48:54.382] [ERROR] error - TypeError: dao.getEmployeesData is not a function
    at ApiEmployee.getEmployeesData (file:///E:/instr-app/instr-app-back/api/employees.js:33:39)
    at getEmployeesData (file:///E:/instr-app/instr-app-back/controllers/employees.js:22:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
[2022-10-31T20:51:43.112] [ERROR] error - TypeError: dao.getShifts is not a function
    at ApiEmployee.getEmployeesData (file:///E:/instr-app/instr-app-back/api/employees.js:35:41)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getEmployeesData (file:///E:/instr-app/instr-app-back/controllers/employees.js:22:29)
[2022-10-31T20:59:46.384] [ERROR] error - TypeError: dao.getShifts is not a function
    at ApiEmployee.getEmployeesData (file:///E:/instr-app/instr-app-back/api/employees.js:35:41)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getEmployeesData (file:///E:/instr-app/instr-app-back/controllers/employees.js:22:29)
[2022-11-01T20:13:40.669] [ERROR] error - Error: routinesschedules validation failed: dueDate: Cast to date failed for value "Invalid Date" (type Date) at path "dueDate"
    at ValidationError.inspect (E:\instr-app\instr-app-back\node_modules\mongoose\lib\error\validation.js:48:26)
    at formatValue (node:internal/util/inspect:806:19)
    at inspect (node:internal/util/inspect:365:10)
    at formatWithOptionsInternal (node:internal/util/inspect:2292:40)
    at Object.format (node:internal/util/inspect:2149:10)
    at basicLayout (E:\instr-app\instr-app-back\node_modules\log4js\lib\layouts.js:63:57)
    at app (E:\instr-app\instr-app-back\node_modules\log4js\lib\appenders\file.js:73:23)
    at E:\instr-app\instr-app-back\node_modules\log4js\lib\log4js.js:43:5
    at Array.forEach (<anonymous>)
    at sendLogEventToAppender (E:\instr-app\instr-app-back\node_modules\log4js\lib\log4js.js:42:21)
    at E:\instr-app\instr-app-back\node_modules\log4js\lib\clustering.js:23:26
    at Array.forEach (<anonymous>)
    at sendToListeners (E:\instr-app\instr-app-back\node_modules\log4js\lib\clustering.js:23:13)
    at Object.send (E:\instr-app\instr-app-back\node_modules\log4js\lib\clustering.js:89:7)
    at Logger._log (E:\instr-app\instr-app-back\node_modules\log4js\lib\logger.js:95:16)
    at Logger.log (E:\instr-app\instr-app-back\node_modules\log4js\lib\logger.js:78:12) {
  errors: {
    dueDate: CastError: Cast to date failed for value "Invalid Date" (type Date) at path "dueDate"
        at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:354:11)
        at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
        at model.$set (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:1380:20)
        at model.$set (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:1122:16)
        at model.Document (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:150:12)
        at model.Model (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:106:12)
        at new model (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:4752:15)
        at Array.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:3318:17)
        at _start (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\parallelLimit.js:27:37)
        at parallelLimit (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\parallelLimit.js:23:5)
        at Function.Model.$__insertMany (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:3348:3)
        at E:\instr-app\instr-app-back\node_modules\kareem\index.js:370:33
        at processTicksAndRejections (node:internal/process/task_queues:78:11) {
      stringValue: '"Invalid Date"',
      messageFormat: undefined,
      kind: 'date',
      value: Invalid Date,
      path: 'dueDate',
      reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
      
        assert.ok(!isNaN(value.valueOf()))
      
          at castDate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\cast\date.js:13:12)
          at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:352:12)
          at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
          at model.$set (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:1380:20)
          at model.$set (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:1122:16)
          at model.Document (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:150:12)
          at model.Model (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:106:12)
          at new model (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:4752:15)
          at Array.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:3318:17)
          at _start (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\parallelLimit.js:27:37)
          at parallelLimit (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\parallelLimit.js:23:5)
          at Function.Model.$__insertMany (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:3348:3)
          at E:\instr-app\instr-app-back\node_modules\kareem\index.js:370:33
          at processTicksAndRejections (node:internal/process/task_queues:78:11) {
        generatedMessage: true,
        code: 'ERR_ASSERTION',
        actual: false,
        expected: true,
        operator: '=='
      },
      valueType: 'Date'
    }
  },
  _message: 'routinesschedules validation failed'
}
[2022-11-01T20:13:40.738] [ERROR] error - Error: routinesschedules validation failed: dueDate: Cast to date failed for value "Invalid Date" (type Date) at path "dueDate"
    at ValidationError.inspect (E:\instr-app\instr-app-back\node_modules\mongoose\lib\error\validation.js:48:26)
    at formatValue (node:internal/util/inspect:806:19)
    at inspect (node:internal/util/inspect:365:10)
    at formatWithOptionsInternal (node:internal/util/inspect:2292:40)
    at Object.format (node:internal/util/inspect:2149:10)
    at basicLayout (E:\instr-app\instr-app-back\node_modules\log4js\lib\layouts.js:63:57)
    at app (E:\instr-app\instr-app-back\node_modules\log4js\lib\appenders\file.js:73:23)
    at E:\instr-app\instr-app-back\node_modules\log4js\lib\log4js.js:43:5
    at Array.forEach (<anonymous>)
    at sendLogEventToAppender (E:\instr-app\instr-app-back\node_modules\log4js\lib\log4js.js:42:21)
    at E:\instr-app\instr-app-back\node_modules\log4js\lib\clustering.js:23:26
    at Array.forEach (<anonymous>)
    at sendToListeners (E:\instr-app\instr-app-back\node_modules\log4js\lib\clustering.js:23:13)
    at Object.send (E:\instr-app\instr-app-back\node_modules\log4js\lib\clustering.js:89:7)
    at Logger._log (E:\instr-app\instr-app-back\node_modules\log4js\lib\logger.js:95:16)
    at Logger.log (E:\instr-app\instr-app-back\node_modules\log4js\lib\logger.js:78:12) {
  errors: {
    dueDate: CastError: Cast to date failed for value "Invalid Date" (type Date) at path "dueDate"
        at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:354:11)
        at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
        at model.$set (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:1380:20)
        at model.$set (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:1122:16)
        at model.Document (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:150:12)
        at model.Model (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:106:12)
        at new model (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:4752:15)
        at Array.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:3318:17)
        at _start (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\parallelLimit.js:27:37)
        at parallelLimit (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\parallelLimit.js:23:5)
        at Function.Model.$__insertMany (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:3348:3)
        at E:\instr-app\instr-app-back\node_modules\kareem\index.js:370:33
        at processTicksAndRejections (node:internal/process/task_queues:78:11) {
      stringValue: '"Invalid Date"',
      messageFormat: undefined,
      kind: 'date',
      value: Invalid Date,
      path: 'dueDate',
      reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
      
        assert.ok(!isNaN(value.valueOf()))
      
          at castDate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\cast\date.js:13:12)
          at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:352:12)
          at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
          at model.$set (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:1380:20)
          at model.$set (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:1122:16)
          at model.Document (E:\instr-app\instr-app-back\node_modules\mongoose\lib\document.js:150:12)
          at model.Model (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:106:12)
          at new model (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:4752:15)
          at Array.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:3318:17)
          at _start (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\parallelLimit.js:27:37)
          at parallelLimit (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\parallelLimit.js:23:5)
          at Function.Model.$__insertMany (E:\instr-app\instr-app-back\node_modules\mongoose\lib\model.js:3348:3)
          at E:\instr-app\instr-app-back\node_modules\kareem\index.js:370:33
          at processTicksAndRejections (node:internal/process/task_queues:78:11) {
        generatedMessage: true,
        code: 'ERR_ASSERTION',
        actual: false,
        expected: true,
        operator: '=='
      },
      valueType: 'Date'
    }
  },
  _message: 'routinesschedules validation failed'
}
[2022-11-01T21:26:56.045] [ERROR] error - TypeError: employees.map is not a function
    at getForScheduleEmployeesDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:9:68)
    at file:///E:/instr-app/instr-app-back/api/employees.js:40:55
    at Array.map (<anonymous>)
    at ApiEmployee.getEmployeesData (file:///E:/instr-app/instr-app-back/api/employees.js:40:39)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getEmployeesData (file:///E:/instr-app/instr-app-back/controllers/employees.js:22:29)
[2022-11-01T21:27:07.800] [ERROR] error - TypeError: employees.map is not a function
    at getForScheduleEmployeesDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:9:68)
    at file:///E:/instr-app/instr-app-back/api/employees.js:40:55
    at Array.map (<anonymous>)
    at ApiEmployee.getEmployeesData (file:///E:/instr-app/instr-app-back/api/employees.js:40:39)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getEmployeesData (file:///E:/instr-app/instr-app-back/controllers/employees.js:22:29)
[2022-11-01T21:32:49.820] [ERROR] error - TypeError: employees.map is not a function
    at employeesDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:15:65)
    at file:///E:/instr-app/instr-app-back/api/employees.js:41:55
    at Array.map (<anonymous>)
    at ApiEmployee.getEmployeesData (file:///E:/instr-app/instr-app-back/api/employees.js:41:39)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getEmployeesData (file:///E:/instr-app/instr-app-back/controllers/employees.js:22:29)
[2022-11-01T21:57:28.700] [ERROR] error - TypeError: stringDate.split is not a function
    at parseStringToDate (file:///E:/instr-app/instr-app-back/utils/formatDate.js:6:26)
    at file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:28:15
    at Array.map (<anonymous>)
    at employeesDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:17:65)
    at ApiEmployee.getEmployeesData (file:///E:/instr-app/instr-app-back/api/employees.js:40:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getEmployeesData (file:///E:/instr-app/instr-app-back/controllers/employees.js:22:29)
[2022-11-01T21:58:13.864] [ERROR] error - TypeError: stringDate.split is not a function
    at parseStringToString (file:///E:/instr-app/instr-app-back/utils/formatDate.js:16:34)
    at file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:28:15
    at Array.map (<anonymous>)
    at employeesDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:17:65)
    at ApiEmployee.getEmployeesData (file:///E:/instr-app/instr-app-back/api/employees.js:40:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getEmployeesData (file:///E:/instr-app/instr-app-back/controllers/employees.js:22:29)
[2022-11-15T21:38:18.133] [ERROR] error - CastError: Cast to Number failed for value "dailyShift" (type string) at path "shift"
    at model.Query.exec (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4540:21)
    at model.Query.Query.then (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4639:15)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  messageFormat: undefined,
  stringValue: '"dailyShift"',
  kind: 'Number',
  value: 'dailyShift',
  path: 'shift',
  reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
  
    assert.ok(!isNaN(val))
  
      at castNumber (E:\instr-app\instr-app-back\node_modules\mongoose\lib\cast\number.js:28:10)
      at SchemaNumber.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\number.js:378:12)
      at SchemaNumber.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
      at SchemaNumber.SchemaType._castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1553:15)
      at SchemaNumber.castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\number.js:432:14)
      at SchemaNumber.SchemaType.castForQueryWrapper (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1520:20)
      at castUpdateVal (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:523:19)
      at walkUpdatePath (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:343:24)
      at castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:95:7)
      at model.Query._castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4749:10)
      at castDoc (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4777:18)
      at model.Query._updateThunk (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:3934:20)
      at model.Query.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4036:23)
      at model.Query._wrappedThunk [as _updateOne] (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\wrapThunk.js:27:8)
      at E:\instr-app\instr-app-back\node_modules\kareem\index.js:279:20
      at _next (E:\instr-app\instr-app-back\node_modules\kareem\index.js:103:16) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: '=='
  },
  valueType: 'string'
}
[2022-11-16T20:51:25.644] [ERROR] error - CastError: Cast to date failed for value "16/8/1983, 21:00:00" (type string) at path "hireDate"
    at model.Query.exec (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4540:21)
    at model.Query.Query.then (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4639:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  messageFormat: undefined,
  stringValue: '"16/8/1983, 21:00:00"',
  kind: 'date',
  value: '16/8/1983, 21:00:00',
  path: 'hireDate',
  reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
  
    assert.ok(false)
  
      at castDate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\cast\date.js:40:10)
      at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:352:12)
      at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
      at SchemaDate.SchemaType._castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1553:15)
      at SchemaDate.castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:387:17)
      at SchemaDate.SchemaType.castForQueryWrapper (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1520:20)
      at castUpdateVal (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:523:19)
      at walkUpdatePath (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:343:24)
      at castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:95:7)
      at model.Query._castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4749:10)
      at castDoc (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4777:18)
      at model.Query._updateThunk (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:3934:20)
      at model.Query.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4036:23)
      at model.Query._wrappedThunk [as _updateOne] (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\wrapThunk.js:27:8)
      at E:\instr-app\instr-app-back\node_modules\kareem\index.js:279:20
      at _next (E:\instr-app\instr-app-back\node_modules\kareem\index.js:103:16) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: '=='
  },
  valueType: 'string'
}
[2022-11-16T20:51:59.183] [ERROR] error - CastError: Cast to date failed for value "25/4/1983, 21:00:00" (type string) at path "hireDate"
    at model.Query.exec (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4540:21)
    at model.Query.Query.then (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4639:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  messageFormat: undefined,
  stringValue: '"25/4/1983, 21:00:00"',
  kind: 'date',
  value: '25/4/1983, 21:00:00',
  path: 'hireDate',
  reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
  
    assert.ok(false)
  
      at castDate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\cast\date.js:40:10)
      at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:352:12)
      at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
      at SchemaDate.SchemaType._castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1553:15)
      at SchemaDate.castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:387:17)
      at SchemaDate.SchemaType.castForQueryWrapper (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1520:20)
      at castUpdateVal (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:523:19)
      at walkUpdatePath (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:343:24)
      at castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:95:7)
      at model.Query._castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4749:10)
      at castDoc (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4777:18)
      at model.Query._updateThunk (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:3934:20)
      at model.Query.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4036:23)
      at model.Query._wrappedThunk [as _updateOne] (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\wrapThunk.js:27:8)
      at E:\instr-app\instr-app-back\node_modules\kareem\index.js:279:20
      at _next (E:\instr-app\instr-app-back\node_modules\kareem\index.js:103:16) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: '=='
  },
  valueType: 'string'
}
[2022-11-16T20:53:57.166] [ERROR] error - CastError: Cast to date failed for value "28/4/1983, 21:00:00" (type string) at path "hireDate"
    at model.Query.exec (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4540:21)
    at model.Query.Query.then (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4639:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  messageFormat: undefined,
  stringValue: '"28/4/1983, 21:00:00"',
  kind: 'date',
  value: '28/4/1983, 21:00:00',
  path: 'hireDate',
  reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
  
    assert.ok(false)
  
      at castDate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\cast\date.js:40:10)
      at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:352:12)
      at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
      at SchemaDate.SchemaType._castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1553:15)
      at SchemaDate.castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:387:17)
      at SchemaDate.SchemaType.castForQueryWrapper (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1520:20)
      at castUpdateVal (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:523:19)
      at walkUpdatePath (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:343:24)
      at castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:95:7)
      at model.Query._castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4749:10)
      at castDoc (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4777:18)
      at model.Query._updateThunk (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:3934:20)
      at model.Query.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4036:23)
      at model.Query._wrappedThunk [as _updateOne] (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\wrapThunk.js:27:8)
      at E:\instr-app\instr-app-back\node_modules\kareem\index.js:279:20
      at _next (E:\instr-app\instr-app-back\node_modules\kareem\index.js:103:16) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: '=='
  },
  valueType: 'string'
}
[2022-11-16T20:55:26.007] [ERROR] error - CastError: Cast to date failed for value "28/4/1983, 21:00:00" (type string) at path "hireDate"
    at model.Query.exec (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4540:21)
    at model.Query.Query.then (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4639:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  messageFormat: undefined,
  stringValue: '"28/4/1983, 21:00:00"',
  kind: 'date',
  value: '28/4/1983, 21:00:00',
  path: 'hireDate',
  reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
  
    assert.ok(false)
  
      at castDate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\cast\date.js:40:10)
      at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:352:12)
      at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
      at SchemaDate.SchemaType._castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1553:15)
      at SchemaDate.castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:387:17)
      at SchemaDate.SchemaType.castForQueryWrapper (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1520:20)
      at castUpdateVal (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:523:19)
      at walkUpdatePath (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:343:24)
      at castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:95:7)
      at model.Query._castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4749:10)
      at castDoc (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4777:18)
      at model.Query._updateThunk (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:3934:20)
      at model.Query.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4036:23)
      at model.Query._wrappedThunk [as _updateOne] (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\wrapThunk.js:27:8)
      at E:\instr-app\instr-app-back\node_modules\kareem\index.js:279:20
      at _next (E:\instr-app\instr-app-back\node_modules\kareem\index.js:103:16) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: '=='
  },
  valueType: 'string'
}
[2022-11-16T20:56:01.251] [ERROR] error - CastError: Cast to date failed for value "28/4/1983, 21:00:00" (type string) at path "hireDate"
    at model.Query.exec (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4540:21)
    at model.Query.Query.then (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4639:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  messageFormat: undefined,
  stringValue: '"28/4/1983, 21:00:00"',
  kind: 'date',
  value: '28/4/1983, 21:00:00',
  path: 'hireDate',
  reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
  
    assert.ok(false)
  
      at castDate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\cast\date.js:40:10)
      at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:352:12)
      at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
      at SchemaDate.SchemaType._castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1553:15)
      at SchemaDate.castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:387:17)
      at SchemaDate.SchemaType.castForQueryWrapper (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1520:20)
      at castUpdateVal (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:523:19)
      at walkUpdatePath (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:343:24)
      at castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:95:7)
      at model.Query._castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4749:10)
      at castDoc (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4777:18)
      at model.Query._updateThunk (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:3934:20)
      at model.Query.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4036:23)
      at model.Query._wrappedThunk [as _updateOne] (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\wrapThunk.js:27:8)
      at E:\instr-app\instr-app-back\node_modules\kareem\index.js:279:20
      at _next (E:\instr-app\instr-app-back\node_modules\kareem\index.js:103:16) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: '=='
  },
  valueType: 'string'
}
[2022-11-16T21:11:34.026] [ERROR] error - CastError: Cast to date failed for value "Invalid Date" (type Date) at path "hireDate"
    at model.Query.exec (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4540:21)
    at model.Query.Query.then (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4639:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  messageFormat: undefined,
  stringValue: '"Invalid Date"',
  kind: 'date',
  value: Invalid Date,
  path: 'hireDate',
  reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
  
    assert.ok(!isNaN(value.valueOf()))
  
      at castDate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\cast\date.js:13:12)
      at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:352:12)
      at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
      at SchemaDate.SchemaType._castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1553:15)
      at SchemaDate.castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:387:17)
      at SchemaDate.SchemaType.castForQueryWrapper (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1520:20)
      at castUpdateVal (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:523:19)
      at walkUpdatePath (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:343:24)
      at castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:95:7)
      at model.Query._castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4749:10)
      at castDoc (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4777:18)
      at model.Query._updateThunk (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:3934:20)
      at model.Query.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4036:23)
      at model.Query._wrappedThunk [as _updateOne] (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\wrapThunk.js:27:8)
      at E:\instr-app\instr-app-back\node_modules\kareem\index.js:279:20
      at _next (E:\instr-app\instr-app-back\node_modules\kareem\index.js:103:16) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: '=='
  },
  valueType: 'Date'
}
[2022-11-16T21:11:54.859] [ERROR] error - CastError: Cast to date failed for value "Invalid Date" (type Date) at path "hireDate"
    at model.Query.exec (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4540:21)
    at model.Query.Query.then (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4639:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  messageFormat: undefined,
  stringValue: '"Invalid Date"',
  kind: 'date',
  value: Invalid Date,
  path: 'hireDate',
  reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
  
    assert.ok(!isNaN(value.valueOf()))
  
      at castDate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\cast\date.js:13:12)
      at SchemaDate.cast (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:352:12)
      at SchemaDate.SchemaType.applySetters (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1122:12)
      at SchemaDate.SchemaType._castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1553:15)
      at SchemaDate.castForQuery (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schema\date.js:387:17)
      at SchemaDate.SchemaType.castForQueryWrapper (E:\instr-app\instr-app-back\node_modules\mongoose\lib\schematype.js:1520:20)
      at castUpdateVal (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:523:19)
      at walkUpdatePath (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:343:24)
      at castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\castUpdate.js:95:7)
      at model.Query._castUpdate (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4749:10)
      at castDoc (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4777:18)
      at model.Query._updateThunk (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:3934:20)
      at model.Query.<anonymous> (E:\instr-app\instr-app-back\node_modules\mongoose\lib\query.js:4036:23)
      at model.Query._wrappedThunk [as _updateOne] (E:\instr-app\instr-app-back\node_modules\mongoose\lib\helpers\query\wrapThunk.js:27:8)
      at E:\instr-app\instr-app-back\node_modules\kareem\index.js:279:20
      at _next (E:\instr-app\instr-app-back\node_modules\kareem\index.js:103:16) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: '=='
  },
  valueType: 'Date'
}
[2022-11-16T21:12:54.344] [ERROR] error - ReferenceError: parseStringToString is not defined
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:131:27)
    at updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
[2022-11-16T21:13:07.429] [ERROR] error - ReferenceError: parseStringToString is not defined
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:131:27)
    at updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
[2022-11-16T21:14:43.641] [ERROR] error - ReferenceError: parseStringToDate is not defined
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:128:34)
    at updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
[2022-11-16T21:15:20.775] [ERROR] error - TypeError: stringDate.split is not a function
    at parseStringToDate (file:///E:/instr-app/instr-app-back/utils/formatDate.js:6:26)
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:128:44)
    at updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
[2022-11-16T21:25:50.952] [ERROR] error - ReferenceError: date is not defined
    at updateEmployeeDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:33:64)
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:137:56)
    at updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
[2022-11-16T21:26:25.030] [ERROR] error - ReferenceError: date is not defined
    at updateEmployeeDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:33:64)
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:137:33)
    at updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
[2022-11-16T21:26:49.304] [ERROR] error - ReferenceError: date is not defined
    at updateEmployeeDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:33:64)
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:137:33)
    at updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
[2022-11-16T21:27:47.992] [ERROR] error - TypeError: dateInLocalDateString(...).getDate is not a function
    at updateEmployeeDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:33:105)
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:137:33)
    at updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
[2022-11-16T21:28:22.382] [ERROR] error - TypeError: dateInLocalDateString(...).getDate is not a function
    at updateEmployeeDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:33:82)
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:137:33)
    at updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
[2022-11-19T11:48:10.085] [ERROR] error - ReferenceError: conose is not defined
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:132:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:29)
[2022-11-19T11:48:15.535] [ERROR] error - ReferenceError: conose is not defined
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:132:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:29)
[2022-11-19T11:51:22.461] [ERROR] error - TypeError: empResp.map is not a function
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:132:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:29)
[2022-11-19T11:52:05.184] [ERROR] error - TypeError: empResp.map is not a function
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:133:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:29)
[2022-11-19T12:06:00.810] [ERROR] error - TypeError: Cannot read properties of undefined (reading 'name')
    at file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:26:76
    at Array.map (<anonymous>)
    at employeesDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:17:65)
    at ApiEmployee.getEmployeesData (file:///E:/instr-app/instr-app-back/api/employees.js:41:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:132:33)
    at async updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:29)
[2022-11-19T12:06:20.236] [ERROR] error - TypeError: Cannot read properties of undefined (reading 'name')
    at file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:26:76
    at Array.map (<anonymous>)
    at employeesDTO (file:///E:/instr-app/instr-app-back/model/DTOs/employee.js:17:65)
    at ApiEmployee.getEmployeesData (file:///E:/instr-app/instr-app-back/api/employees.js:41:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getEmployeesData (file:///E:/instr-app/instr-app-back/controllers/employees.js:22:29)
[2022-11-19T12:12:33.394] [ERROR] error - ReferenceError: Cannot access 'employee' before initialization
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:129:25)
    at updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:52)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (E:\instr-app\instr-app-back\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
    at Function.handle (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:174:3)
    at router (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:47:12)
    at Layer.handle [as handle_request] (E:\instr-app\instr-app-back\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:317:13)
    at E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:335:12)
    at next (E:\instr-app\instr-app-back\node_modules\express\lib\router\index.js:275:10)
[2022-11-19T12:18:32.064] [ERROR] error - ReferenceError: hireDate is not defined
    at DBMongoDao.updateEmployee (file:///E:/instr-app/instr-app-back/model/DAOs/DBMongoDao.js:137:35)
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:133:41)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:29)
[2022-11-19T12:19:19.924] [ERROR] error - ReferenceError: hireDate is not defined
    at DBMongoDao.updateEmployee (file:///E:/instr-app/instr-app-back/model/DAOs/DBMongoDao.js:137:35)
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:133:41)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:29)
[2022-11-19T12:19:45.563] [ERROR] error - ReferenceError: hireDate is not defined
    at DBMongoDao.updateEmployee (file:///E:/instr-app/instr-app-back/model/DAOs/DBMongoDao.js:137:35)
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:133:41)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:29)
[2022-11-19T12:20:37.692] [ERROR] error - ReferenceError: hireDate is not defined
    at DBMongoDao.updateEmployee (file:///E:/instr-app/instr-app-back/model/DAOs/DBMongoDao.js:137:35)
    at ApiEmployee.updateEmployee (file:///E:/instr-app/instr-app-back/api/employees.js:133:41)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async updateEmployee (file:///E:/instr-app/instr-app-back/controllers/employees.js:48:29)
